### ISSUE TITLE
entityViewManager.save doesn't respect @EntityViewInheritance

### ISSUE THREAD

**Issue by david-kubecka** [Blazebit/blaze-persistence#1656]
### Description

tl;dr: `entityViewManager.save` wrongly downgrades the actual entity view type to a base type, even when there's proper entity/entity view inheritance set up. This leads to issues e.g. in the GraphQL context where base interfaces are used for modeling union types.

Entity setup:
```
@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
class Animal {
    @OneToOne(fetch = FetchType.LAZY)
    var owner: Person? = null
}

@Entity
class Cat(
    var name: String,
) : Animal()

@Entity
class Person(
    val name: String,

    @OneToOne(fetch = FetchType.LAZY, mappedBy = "owner")
    val animal: Animal? = null,
)
```

Entity view setup:
```
@CreatableEntityView
@EntityView(Animal::class)
@EntityViewInheritance
interface AnimalView {
    @get:IdMapping
    val id: Long?
}

@CreatableEntityView
@EntityView(Cat::class)
interface CatView : AnimalView {
    var name: String
}

@CreatableEntityView
@EntityView(Person::class)
interface PersonView {
    @get:IdMapping
    val id: Long?
    var name: String
    var animal: AnimalView
}
```

Service method:
```
    @Transactional
    fun createPerson(person: PersonView): PersonView {
        entityViewManager.save(entityManager, person)
        return person
    }
```

### Expected behavior

If a concrete PersonView with CatView animal is passed to `createPerson` then the same `PersonView` is returned.

### Actual behavior

The CatView is downgraded to AnimalView, probably as per the last paragraph in https://persistence.blazebit.com/documentation/1.6/entity-view/manual/en_US/#create-mapping. 

### Environment

BP: 1.6.8
